### ISSUE TITLE
Create Version failed. Bad model detected to create keras h5 model with custom model prediction

### ISSUE THREAD

**Issue by cah-aswini-jalla** [GoogleCloudPlatform/python-docs-samples#5711]
I was trying to deploy a keras model using custom model prediction code as below but facing issue while deploying the model 

**Error:**
Create Version failed. Bad model detected with error: "Failed to load model: User-provided package pd-package-0.1.tar.gz failed to install: Command '['python-default', '-m', 'pip', 'install', '--target=/tmp/custom_lib', '--no-cache-dir', '-b', '/tmp/pip_builds', '/tmp/custom_code/pd-package-0.1.tar.gz']' returned non-zero exit status 1. (Error code: 0)"

**Deployment Code:**
model_id = 'projects/{}/models/{}'.format(str(gcp_project_name), str(modelname))
request_version_dict = {
                                'name': 'v1', 'deployment_uri': deploymentpath,
                                'predictionClass': 'pd_model_prediction.CustomModelPrediction',
                                'runtimeVersion': '2.4', 'pythonVersion': '3.7',
                                'autoScaling': {'minNodes': 1}, 'packageUris': packagepath
}

service.projects().models().versions().create(parent=model_id, body=request_version_dict).execute()

**Keras Classifier Model Deployment:**
        with open(model_filename, 'wb') as f:
            classifier_estimator_all_data.model.save(model_filename, f)

**Setup File:**
from setuptools import setup

setup(
  name="pd-package",
  version="0.1",
  include_package_data=True,
  scripts=["pd_preprocess.py", "pd_model_prediction.py"]
)

run_setup('setup.py', script_args=['sdist'])

**Custom Model Prediction File:**
import os
import pickle
import json
import numpy as np

PRE_PROCESS_PICKLE_NAME='pd_processor_state.pkl'
MODEL_FILENAME='pd_email_class.h5'

class CustomModelPrediction(object):

    def __init__(self, model, processor):
        self._model = model
        self._processor = processor

    def _postprocess(self, predictions):
        labels = ['Account/Inquiry', 'Billing / Invoice', 'Credits', 'Delivery', 'Drop Ship Order', 'Emergency Order',
          'General Inquiry', 'Off Schedule Order', 'Order Discrepancy', 'Order Processing', 'Pricing', 'Product Inquiry',
          'Product Quality', 'Program / Promotions', 'Returns']
        label_indexes = [np.argmax(prediction) for prediction in predictions]
        return [labels[label_index] for label_index in label_indexes]

    def predict(self, instances, **kwargs):
        preprocessed_data = self._processor.transform(instances)
        preprocessed_data.sort_indices()
        predictions = self._model.predict(preprocessed_data)
        classes = predictions.argmax(axis=1)
        labels = self._postprocess(predictions)
        prob = self._model.predict(preprocessed_data)
        max_probability = prob.max(1)
        pred_with_prob = np.column_stack((labels, max_probability))
        pred_json = json.dumps({'Prediction': pred_with_prob.tolist()})
        return pred_json

    @classmethod
    def from_path(cls, model_dir):
        import tensorflow.keras as keras
        model = keras.models.load_model(
          os.path.join(model_dir, MODEL_FILENAME))
        with open(os.path.join(model_dir, PRE_PROCESS_PICKLE_NAME), 'rb') as f:
        #with open(os.path.join(model_dir,'./vectorizer.pickle'), 'rb') as a
        #    and open(os.path.join(model_dir,'./selector.pickle'), 'rb') as b:
            processor = pickle.load(f)

        return cls(model, processor)

**Note:** Executing all above code using PyCharm

Python Version: 3.7.9
Keras Version: 2.2.4
Tensorflow Version: 2.3.1
Sklearn Version: 0.24.1

Thanks.
